CMakeLists.txt

```diff
@@ -32,6 +32,11 @@ if(CMAKE_C_COMPILER_ID MATCHES "TI")
   set(USING_TI 1)
 endif()
 
+# Try to determine compiler version if not automatically set by CMake
+if(USING_GCC AND NOT CMAKE_C_COMPILER_VERSION)
+  EXECUTE_PROCESS(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE CMAKE_C_COMPILER_VERSION)
```

Comment:
(the previous thread became hidden, so let's continue here)

I have serious doubts that this is a proper way to go. Actually, I have serious doubts about the whole 'external' toolchain file and the `CMAKE_FORCE_C_COMPILER`. If one can pass `-DCMAKE_TOOLCHAIN_FILE=cmake/toolchain_external.cmake -DEXTERNAL_CMAKE_SYSTEM_PROCESSOR=armv7l -DEXTERNAL_CMAKE_C_COMPILER=arm-none-eabi-gcc -DEXTERNAL_CMAKE_C_COMPILER_ID=GNU` to cmake, why cannot they simply pass `-DCMAKE_SYSTEM_NAME=MCU -DCMAKE_SYSTEM_PROCESSOR=armv7l -DCMAKE_C_COMPILER=arm-none-eabi-gcc`? If I'm not mistaken, it is not mandatory to specify a toolchain file. And `CMAKE_C_COMPILER` can also be set without 'forcing' it (see https://cmake.org/Wiki/CMake_FAQ#How_do_I_use_a_different_compiler.3F). And then, I guess, `CMAKE_C_COMPILER_VERSION` would be correctly detected without any workarounds.

Reply:
I was thinking of something like this: https://github.com/akosthekiss/jerryscript/commit/1534146857d5f1b7b835d270a7ed6ed9c278002e . (This is speculative though as I cannot validate these changes.)

Reply:
I agree, it looks better. I'm going to close this then.